src/ray/core_worker/transport/dependency_resolver.h

```diff
@@ -99,10 +98,14 @@ class LocalDependencyResolver {
   TaskFinisherInterface &task_finisher_;
 
   ActorCreatorInterface &actor_creator_;
-  /// Number of tasks pending dependency resolution.
-  std::atomic<int> num_pending_;
 
-  absl::flat_hash_map<TaskID, std::unique_ptr<TaskState>> pending_tasks_
+  // ID of the dependency resolution request. Used to index pending_tasks_ because a
+  // same TaskID may be requested multiple times and we treat each request independently.
+  using RequestID = size_t;
+
+  RequestID next_request_id_ ABSL_GUARDED_BY(mu_) = 0;
+
+  absl::flat_hash_map<RequestID, std::unique_ptr<TaskState>> pending_tasks_
```

Comment:
Can we make it `absl::flat_hash_map<TaskID, std::vector<std::unique_ptr<TaskState>> >pending_tasks_`

Reply:
done